internal/api/grpc/user/v2/query_integration_test.go

```diff
@@ -0,0 +1,629 @@
+		name    string
+		args    args
+		want    *user.GetUserByIDResponse
+		wantErr bool
+	}{
+		{
+			name: "user by ID, no id provided",
+			args: args{
+				IamCTX,
+				&user.GetUserByIDRequest{
+					Organization: &object.Organization{
+						Org: &object.Organization_OrgId{
+							OrgId: Tester.Organisation.ID,
+						},
+					},
+					UserId: "",
+				},
+				func(ctx context.Context, username string, request *user.GetUserByIDRequest) error {
+					return nil
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "user by ID, not found",
+			args: args{
+				IamCTX,
+				&user.GetUserByIDRequest{
+					Organization: &object.Organization{
+						Org: &object.Organization_OrgId{
+							OrgId: Tester.Organisation.ID,
+						},
+					},
+					UserId: "unknown",
+				},
+				func(ctx context.Context, username string, request *user.GetUserByIDRequest) error {
+					return nil
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "user by ID, ok",
+			args: args{
+				IamCTX,
+				&user.GetUserByIDRequest{
+					Organization: &object.Organization{
+						Org: &object.Organization_OrgId{
+							OrgId: Tester.Organisation.ID,
+						},
+					},
+				},
+				func(ctx context.Context, username string, request *user.GetUserByIDRequest) error {
+					resp := Tester.CreateHumanUserVerified(ctx, orgResp.OrganizationId, username)
+					request.UserId = resp.GetUserId()
+					return nil
+				},
+			},
+			want: &user.GetUserByIDResponse{
+				User: &user.User{
+					State:              user.UserState_USER_STATE_ACTIVE,
+					Username:           "",
+					LoginNames:         nil,
+					PreferredLoginName: "",
+					Type: &user.User_Human{
+						Human: &user.HumanUser{
+							Profile: &user.HumanProfile{
+								GivenName:         "Mickey",
+								FamilyName:        "Mouse",
+								NickName:          gu.Ptr("Mickey"),
+								DisplayName:       gu.Ptr("Mickey Mouse"),
+								PreferredLanguage: gu.Ptr("nl"),
+								Gender:            user.Gender_GENDER_MALE.Enum(),
+								AvatarUrl:         "",
+							},
+							Email: &user.HumanEmail{
+								IsVerified: true,
+							},
+							Phone: &user.HumanPhone{
+								Phone:      "+41791234567",
+								IsVerified: true,
+							},
+						},
+					},
+				},
+				Details: &object.Details{
+					ChangeDate:    timestamppb.Now(),
+					ResourceOwner: orgResp.OrganizationId,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			username := fmt.Sprintf("%d@mouse.com", time.Now().UnixNano())
+			err := tt.args.dep(tt.args.ctx, username, tt.args.req)
+			require.NoError(t, err)
+
+			var got *user.GetUserByIDResponse
+			for {
+				got, err = Client.GetUserByID(tt.args.ctx, tt.args.req)
+				if err == nil || (tt.wantErr && err != nil) {
+					break
+				}
+				select {
+				case <-CTX.Done():
+					t.Fatal(CTX.Err(), err)
+				case <-time.After(time.Second):
+					t.Log("retrying GetUserByID")
+					continue
+				}
+			}
+			if tt.wantErr {
```

Comment:
ByID queries are not eventually consistent, right? Else, can we use require.EventuallyWithT

Reply:
They are I think, as it is an integration test you first with either an API call or command function which still has to be processed for the projection and then the query side, or am I wrong?

Reply:
I think if ByID queries should not be eventually consistent, we should not await eventual consistency (which would by design be a bug then).

Reply:
Why should we not await eventual consistency? I mean as with CQRS and no direct change in the projections all read operations are eventual consistent, which should be included in the integration test right? Maybe we could include a timeout for "performance" sake but we always have to expect that we don't have it instantly on the read side

Reply:
We should discuss the ByID topic in another frame, I think.